File: 010_sparse_matrices/solution/four_corners.m
function A = four_corners(m,n)
%FOUR_CORNERS Construct a m x n sparse matrix with ones in each of its four
%corners.
%
% t = four_corners(m,n);
%
% Inputs:
%  m,n  the dimension of the matrix requested
% Outputs:
%  A  the sparse matrix with ones in their four corners
%

A = sparse(m,n);
A(1,1) = 1;
A(m,1) = 1;
A(1,n) = 1;
A(m,n) = 1;

end

File: 010_sparse_matrices/solution/four_corners.py
import numpy as np, scipy as sp
def four_corners(m,n):
    """This function constructs an m x n sparse matrix with ones in each of its
    four corners.
    """

    i = np.array([0, 0, m-1, m-1])
    j = np.array([0, n-1, 0, n-1])
    k = np.array([1., 1., 1., 1.])
    A = sp.sparse.csr_matrix((k,(i,j)), shape=(m,n))

    return A


